rocketpy/simulation/flight.py

```diff
@@ -1053,6 +726,573 @@ def __simulate(self, verbose):
+            The current flight phase.
+        """
+        if self.sensors:
+            u_dot = phase.derivative(self.t, self.y_sol)
+            self.__measure_sensors(node._component_sensors, u_dot)
+
+        for controller in node._controllers:
+            controller(
+                self.t,
+                self.y_sol,
+                self.solution,
+                self.sensors,
+            )
+
+    def __measure_sensors(self, component_sensors, u_dot, t=None, y_sol=None):
+        """Measure sensors with the given state and derivative.
+
+        Parameters
+        ----------
+        component_sensors : list
+            List of (sensor, position) tuples.
+        u_dot : array_like
+            State derivative vector.
+        t : float, optional
+            Time for measurement. If None, uses self.t.
+        y_sol : array_like, optional
+            State vector. If None, uses self.y_sol.
+        """
+        if t is None:
+            t = self.t
+        if y_sol is None:
+            y_sol = self.y_sol
+
+        for sensor, position in component_sensors:
+            relative_position = position - self.rocket._csys * Vector(
+                [0, 0, self.rocket.center_of_dry_mass_position]
+            )
+            sensor.measure(
+                t,
+                u=y_sol,
+                u_dot=u_dot,
+                relative_position=relative_position,
+                environment=self.env,
+                gravity=self.env.gravity.get_value_opt(
+                    y_sol[2] if len(y_sol) > 2 else self.solution[-1][3]
+                ),
+                pressure=self.env.pressure,
+                earth_radius=self.env.earth_radius,
+                initial_coordinates=(self.env.latitude, self.env.longitude),
+            )
+
+    def __check_and_handle_parachute_triggers(
+        self, node, phase, phase_index, node_index
+    ):
+        """Check for parachute triggers and handle deployment.
+
+        Parameters
+        ----------
+        node : TimeNode
+            The current time node.
+        phase : FlightPhase
+            The current flight phase.
+        phase_index : int
+            The index of the current phase.
+        node_index : int
+            The index of the current node.
+
+        Returns
+        -------
+        bool
+            True if a parachute was triggered and the phase should break.
+        """
+        for parachute in node.parachutes:
+            # Calculate and save pressure signal
+            (
+                noisy_pressure,
+                height_above_ground_level,
+            ) = self.__calculate_and_save_pressure_signals(
+                parachute, node.t, self.y_sol[2]
+            )
+            if not parachute.triggerfunc(
+                noisy_pressure,
+                height_above_ground_level,
+                self.y_sol,
+                self.sensors,
+            ):
+                return False  # Early exit: parachute not deployed
```

Comment:
Really not sure, since the code is complex, but shouldn't this be a `continue` (i.e. even if one trigger does not go off, there other parachutes to check)?

Reply:
Yeah, if there is more than one parachute here the pressure signals wont be logged correctly